src/cryptsetup/cryptsetup-generator.c

```diff
@@ -37,9 +38,54 @@ static Hashmap *arg_disks = NULL;
 static char *arg_default_options = NULL;
 static char *arg_default_keyfile = NULL;
 
+static int generate_keydev_mount(const char *name, const char *keydev, char **unit, char **mount) {
+        _cleanup_free_ char *u = NULL, *what = NULL, *where = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        int r;
+
+        assert(name);
+        assert(keydev);
+        assert(unit);
+        assert(mount);
+
+        where = strjoin("/run/systemd/cryptsetup/keydev-", name);
+        if (!where)
+                return -ENOMEM;
```

Comment:
Shouldn't this path be readable by root only? People use e.g. vfat usb-sticks for this, and then the key would be publicly visible. Since it's easy to do so, I'd just insert a directory with 0700 mode in between, e.g. `/run/systemd/cryptsetup`.

Reply:
It's trickier to modify the permissions of the mount point (the directory that is mounted over itself doesn't matter, the permissions that are in the root directory of the mounted fs matter). It might be possible to get this to work, but the usual solution is to use on of the intermediate directories. If you think `/run/systmed/cryptsetup` is not the right place, then one more intermediate level should be inserted: `/run/systemd/cryptsetup/private/...`.

Reply:
Right, right...I've set 0700 for ```/run/systemd/cryptsetup```